hub/gads-ui/src/components/DeviceControl/StreamCanvas/StreamCanvas.js

```diff
@@ -45,6 +45,54 @@ export default function StreamCanvas({ deviceData, shouldShowStream }) {
         streamUrl = `/device/${deviceData.udid}/android-stream-mjpeg`
     }
 
+    useEffect(() => {
+        function handleKeyDown(event) {
+            // Check if there is an active element to avoid capturing keyboard events
+            // When we want to be typing somewhere else, not send to the device
+            const activeElement = document.activeElement
+            const isInputFocused = activeElement && (
+                activeElement.tagName === 'INPUT' ||
+                activeElement.tagName === 'TEXTAREA' ||
+                activeElement.isContentEditable
+            )
+
+            // Don't process keyboard events when user is typing somewhere
+            if (isInputFocused) {
+                return
```

Comment:
Consider adding a mechanism to log or track ignored events for debugging purposes. This would help identify unintended behavior if isInputFocused conditions are misinterpreted.

Reply:
Done [here](https://github.com/shamanec/GADS/pull/185/commits/4957832b1908943c42adb380c3ebcb9b9852e2b1)